keras_cv/backend/__init__.py

```diff
@@ -11,91 +11,28 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""
-Keras backend module.
-
-This module adds a temporarily Keras API surface that is fully under KerasCV
-control. This allows us to switch between `keras_core` and `tf.keras`, as well
-as add shims to support older version of `tf.keras`.
-- `config`: check which backend is being run.
-- `keras`: The full `keras` API (via `keras_core` or `tf.keras`).
-- `ops`: `keras_core.ops`, always tf-backed if using `tf.keras`.
-"""
-
-import types
-
-from packaging.version import parse
-
-from keras_cv.backend.config import multi_backend
-
-# Keys are of the form: "module.where.attr.exists->module.where.to.alias"
-# Value are of the form: ["attr1", "attr2", ...] or
-#                        [("attr1_original_name", "attr1_alias_name")]
-_KERAS_CORE_ALIASES = {
-    "utils->saving": [
-        "register_keras_serializable",
-        "deserialize_keras_object",
-        "serialize_keras_object",
-        "get_registered_object",
-    ],
-    "models->saving": ["load_model"],
-}
-
-
-if multi_backend():
-    import keras
-
-    if not hasattr(keras, "__version__") or parse(keras.__version__) < parse(
-        "3.0"
-    ):
-        import keras_core as keras
-
-    keras.backend.name_scope = keras.name_scope
-else:
-    from tensorflow import keras
-
-    if not hasattr(keras, "saving"):
-        keras.saving = types.SimpleNamespace()
-
-    # add aliases
-    for key, value in _KERAS_CORE_ALIASES.items():
-        src, _, dst = key.partition("->")
-        src = src.split(".")
-        dst = dst.split(".")
-
-        src_mod, dst_mod = keras, keras
-
-        # navigate to where we want to alias the attributes
-        for mod in src:
-            src_mod = getattr(src_mod, mod)
-        for mod in dst:
-            dst_mod = getattr(dst_mod, mod)
-
-        # add an alias for each attribute
-        for attr in value:
-            if isinstance(attr, tuple):
-                src_attr, dst_attr = attr
-            else:
-                src_attr, dst_attr = attr, attr
-            attr_val = getattr(src_mod, src_attr)
-            setattr(dst_mod, dst_attr, attr_val)
-
-    # TF Keras doesn't have this rename.
-    keras.activations.silu = keras.activations.swish
-
 from keras_cv.backend import config  # noqa: E402
+from keras_cv.backend import keras  # noqa: E402
 from keras_cv.backend import ops  # noqa: E402
 from keras_cv.backend import random  # noqa: E402
 from keras_cv.backend import tf_ops  # noqa: E402
 
 
 def assert_tf_keras(src):
-    if multi_backend():
+    if config.multi_backend():
         raise NotImplementedError(
             f"KerasCV component {src} does not yet support Keras Core, and can "
             "only be used in `tf.keras`."
         )
 
 
 def supports_ragged():
```

Comment:
These next three fns seem like they could just live in `config` rather then proxying them here.

```
from keras_cv.backend import config

if config.multi_backend():
    ...

if config.supports_ragged():
   ...
```

Is fairly clear usage throughout the library.

Reply:
updated

Reply:
nit: still think this might be slightly clearer to expose in the `config.py` module, as your are checking the status of feature support, but not a big deal! fine either way

Reply:
done